### ISSUE TITLE
Azure DevOps pipeline breaks as soon as new version of dotnet-coverage tool is published

### ISSUE THREAD

**Issue by Kralizek** [dotnet/core#8844]
### Description

I have the dotnet-coverage tool installed locally and configured in the `dotnet-tool.json` file and I use `dotnet tool restore` to restore all the tools in the pipeline. 

As per the title, as soon as a new version is pushed to NuGet, my builds fail.

I had this issue when the 17.9.1 was pushed and I opened this issue https://github.com/microsoft/azure-pipelines-agent/issues/4472

The issue was solved by changing the dotnet-tool.json to point to the latest version. 

```json
{
  "version": 1,
  "isRoot": true,
  "tools": {
    "dotnet-coverage": {
      "version": "17.9.1",
      "commands": [
        "dotnet-coverage"
      ]
    }
  }
}
```

Today, a new version was pushed, 17.9.3, and the build failed again.

I don't think this is the expected behavior, hence the issue.

### Configuration

I use .NET SDK 8 RC2

### Regression?

It's a new project started after 17.8.6 was pushed to NuGet.

### Other information

Everything works fine on my local machine but fails on Azure DevOps pipeline using default agent and `ubuntu-latest`.

**Comment by liveans** [comment]
@dsplaisted do you have any idea who might be the correct person here? it's related to cli AFAIU.

**Comment by dsplaisted** [comment]
How does it fail?  What error message does it give?

I think this is probably caused by this known issue, which will be fixed for GA: https://github.com/dotnet/core/blob/main/release-notes/8.0/known-issues.md#-80100-rc2-dotnet-tool-restore-will-always-install-the-latest-version-of-a-specific-tool

**Comment by Kralizek** [comment]
@dsplaisted most of the information like logging and how to reproduce the issue is in the issue linked in the opening post

**Comment by dsplaisted** [comment]
Looking at the log, it does appear that this is the known issue in RC2 that I mentioned and should be fixed in the 8.0 GA release.
